doc/classes/StringName.xml

```diff
@@ -8,6 +8,7 @@
 		You will usually just pass a [String] to methods expecting a [StringName] and it will be automatically converted, but you may occasionally want to construct a [StringName] ahead of time with the [StringName] constructor or, in GDScript, the literal syntax [code]&amp;"example"[/code].
 		See also [NodePath], which is a similar concept specifically designed to store pre-parsed scene tree paths.
 		All of [String]'s methods are available in this class too. They convert the [StringName] into a string, and they also return a string. This is highly inefficient and should only be used if the string is desired.
+		[b]Note:[/b] In C#, none of the methods listed on this page are available. Instead, use the provided [code]ToString()[/code] method to cast a [StringName] to a string, and then use the equivalent methods in [code]System.String[/code] or [code]StringExtensions[/code]. For convenience, a list of equivalent methods is provided [url=$DOCS_URL/tutorials/scripting/c_sharp/c_sharp_differences.html#string]here[/url].
```

Comment:
Just a nit pick, but I disagree a bit with the wording of the note about these methods not being available. AFAIK GDScript is just implicitly converting the StringName to String behind your back, so using `ToString` in C# is doing the same thing but more explicitly.

We discussed in [RC](https://chat.godotengine.org/channel/dotnet?msg=hKNuwHB89usJxrKy2) a while ago that StringName should be added to the list of classes with notable differences in C#:

https://github.com/godotengine/godot/blob/179dfdc8d78b5bd5377dd115af026df58308bdaf/doc/tools/make_rst.py#L105

It was an oversight when I added this note in https://github.com/godotengine/godot/pull/79206.

Reply:
Okiedokie. What are you suggesting the note should say then, if anything?

Reply:
How about something like this?

```suggestion
		[b]Note:[/b] In C#, an explicit conversion to [code]System.String[/code] is required to use the methods listed on this page. Use the [code]ToString()[/code] method to cast a [StringName] to a string, and then use the equivalent methods in [code]System.String[/code] or [code]StringExtensions[/code]. For convenience, a list of equivalent methods is provided [url=$DOCS_URL/tutorials/scripting/c_sharp/c_sharp_differences.html#string]here[/url].
```

We probably don't even need the reference to the differences documentation page since it would be in the note added when the `StringName` class is added to the `CLASSES_WITH_CSHARP_DIFFERENCES` list. For reference, the note looks like this:

![image](https://github.com/godotengine/godot/assets/3903059/310e8f34-9bd3-4345-8304-35751b741a16)